docs/api.md

```diff
@@ -71,54 +71,48 @@ _Will be removed in v3.0.0._
 
 ## Configuration
 
-### #component_parent_class
-
-Parent class for generated components
-
-    config.view_component.component_parent_class = "MyBaseComponent"
-
-Defaults to "ApplicationComponent" if defined, "ViewComponent::Base" otherwise.
-
 ### #default_preview_layout
 
 Set a custom default layout used for preview index and individual previews:
 
     config.view_component.default_preview_layout = "component_preview"
 
-### #generate_distinct_locale_files
+### #generate
 
-Always generate as many translations files as available locales:
+Configuration for generators
 
-    config.view_component.generate_distinct_locale_files = true
+Defaults to `false` unless otherwise stated.
 
-Defaults to `false`.
+Always generate a component with a sidecar directory:
 
-One file will be generated for each configured `I18n.available_locales`.
-Fallback on `[:en]` when no available_locales is defined.
+    config.view_component.generate.sidecar = true
```

Comment:
🤔 I wonder if a noun would be more appropriate for this namespace:

`config.view_component.generator.sidecar`

Reply:
I considered that, but I think naming it `generate` makes it quite clear that we're `config`uring whether to `generate` the `sidecar` directory, `distinct_locale_files`, and other things. It makes the responsibility of this area of config explicit - whether to generate certain things. 

Granted there are a couple of things that don't fit that mold, (`view_component_path` and `parent_component`), but that pushes me to wonder whether they should really be part of the namespace. The former example definitely has uses outside of generators. The latter I'm thinking could be renamed `inheriting_from`, so to say you're `generate`ing a component `inheriting_from` `ApplicationComponent`, but there's also the argument that what we have already is clear.

Reply:
With this in mind, I think I should drop ceed17032b57346ce8eff5bf302c834b403ee1f1 and 92e8b834e8f66310ef574f5f338a9bb96b3a95b7. Does that sound correct?

Reply:
I think your argument makes sense. Did you consider `generates`?